### ISSUE TITLE
[BUG] - Dynamic Test Names Generated Inside test.describe Cause an Infinite Loop

### ISSUE THREAD

**Issue by keegandonley** [microsoft/playwright#13939]
**Context:**
- Playwright Version: 1.21.1
- Operating System: Mac
- Node.js version: 15.14.0
- Browser: Chromium
- Extra: M1 CPU

## System:
 - OS: macOS 12.2.1
 - Memory: 120.39 MB / 32.00 GB
## Binaries:
 - Node: 15.14.0 - ~/.nvm/versions/node/v15.14.0/bin/node
 - npm: 7.7.6 - ~/.nvm/versions/node/v15.14.0/bin/npm
## Languages:
 - Bash: 3.2.57 - /bin/bash

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```javascript
test.describe('Test dynamic name', () => {
  const dyn = Date.now();

  test.only(`This is a ${dyn} test`, () => {
    expect(1).toBe(1);
  });
});
```

**Describe the bug**

A test with a randomly changing name, like in the simple above example, causes an infinite loop where `test.describe` is called repeatedly but the test never starts, and doesn't respect the timeout limit. Tests in this format were previously working, and now cause my CI to hang indefinitely

**Comment by ltsuda** [comment]
maybe related to https://github.com/microsoft/playwright/issues/13892 ?

**Comment by keegandonley** [comment]
@ltsuda Yes, it does look to be the same problem as that issue. I'm fine with not having a dynamic test name, however, it feels like causing an infinite loop preventing the tests from exiting isn't the most desirable behavior.

**Comment by dgozman** [comment]
Thank you for the issue! This issue has the same cause as #13611, and was fixed by #13666. Next 1.22 release will have it.
